spec/index.bs

```diff
@@ -8558,31 +8590,62 @@ Issue: Because timestamp query provides high-resolution GPU timestamp, we need t
 
 ## {{HTMLCanvasElement/getContext()|HTMLCanvasElement.getContext()}} ## {#canvas-getcontext}
 
-A {{GPUCanvasContext}} object can be obtained via the {{HTMLCanvasElement/getContext()}}
-method of an {{HTMLCanvasElement}} instance by passing the string literal `'webgpu'` as its
-`contextType` argument.
+A {{GPUCanvasContext}} object can be obtained and reconfigured via the
+{{HTMLCanvasElement/getContext()}} method of an {{HTMLCanvasElement}} instance by passing the string
+literal `'webgpu'` as its `contextType` argument. Calling `getContext('webgl', configuration)` a
+second time will reconfigure the context with the new configuration options that were passed.
 
 <div class="example">
     Get a {{GPUCanvasContext}} from an offscreen {{HTMLCanvasElement}}:
     <pre highlight="js">
         const canvas = document.createElement('canvas');
-        const context = canvas.getContext('webgpu');
-        context.configure(/* ... */);
+        const context = canvas.getContext('webgpu', {
+            device: gpuDevice,
+            format: 'bgra8unorm'
+        });
         // ...
     </pre>
 </div>
 
+{{HTMLCanvasElement}} has the following new internal slots:
+
+<dl dfn-type=attribute dfn-for=GPUCanvasContext>
+    : <dfn>\[[gpu_context]]</dfn> of type {{GPUCanvasContext}}?
+    ::
+        The {{GPUCanvasContext}} associated with this canvas, if any. Initially set to `null`.
+</dl>
+
+<div algorithm>
+    When {{HTMLCanvasElement/getContext()|HTMLCanvasElement.getContext()}} is called on a
+    {{HTMLCanvasElement}} |canvas| with a `contextType` of `'webgpu'`, <dfn abstract-op>create or
+    reconfigure the context</dfn> by performing the following steps:
+
+    1. Let |configuration| be the second option argument {{HTMLCanvasElement/getContext()}} was
+        invoked with as a {{GPUCanvasConfiguration}}?.
+    1. If any of the following conditions are unsatisfied, throw an {{InvalidStateError}} and stop
+        these steps.
+        <div class=validusage>
+            - |configuration| is not `null`.
+            - |configuration|.{{GPUCanvasConfiguration/device}} is [=map/exists|specified=].
+            - |configuration|.{{GPUCanvasConfiguration/format}} is [=map/exists|specified=].
+        </div>
+    1. If |canvas|.{{GPUCanvasContext/[[gpu_context]]}} is `null`:
+        1. Let |canvas|.{{GPUCanvasContext/[[gpu_context]]}} be a new {{GPUCanvasContext}}.
+    1. If |context| be |canvas|.{{GPUCanvasContext/[[gpu_context]]}}.
+    1. [$invalidate the current texture$] of |context|.
+    1. Set |context|.{{GPUCanvasContext/[[configuration]]}} to |configuration|.
+    1. Return |context|.
+</div>
+
 ## GPUCanvasContext ## {#canvas-context}
 
 <script type=idl>
 [Exposed=(Window, DedicatedWorker), SecureContext]
 interface GPUCanvasContext {
     readonly attribute (HTMLCanvasElement or OffscreenCanvas) canvas;
 
-    undefined configure(GPUCanvasConfiguration configuration);
     undefined unconfigure();
```

Comment:
do we still need this? I guess it's a distant relative to `buffer.destroy`.
Without a `configure` method it seems strange though. Maybe it should be called `destroy`?